### PR TITLE
`azurerm_data_factory_dataset_parquet` - Support `azure_blob_fs_location`

### PR DESCRIPTION
**Pull request by bianyifan**
Acceptance test:

```
$ make acctests SERVICE='datafactory' TESTARGS='-run=TestAccDataFactoryDatasetParquet_blobFS' TESTTIMEOUT=60m
==> Checking that code complies with gofmt requirements...
==> Checking that Custom Timeouts are used...
==> Checking that acceptance test packages are used...
TF_ACC=1 go test -v ./internal/services/datafactory -run=TestAccDataFactoryDatasetParquet_blobFS -timeout 60m -ldflags="-X=github.com/hashicorp/terraform-provider-azurerm/version.ProviderVersion=acc"
=== RUN   TestAccDataFactoryDatasetParquet_blobFS
=== PAUSE TestAccDataFactoryDatasetParquet_blobFS
=== RUN   TestAccDataFactoryDatasetParquet_blobFSDynamicPath
=== PAUSE TestAccDataFactoryDatasetParquet_blobFSDynamicPath
=== CONT  TestAccDataFactoryDatasetParquet_blobFS
=== CONT  TestAccDataFactoryDatasetParquet_blobFSDynamicPath
--- PASS: TestAccDataFactoryDatasetParquet_blobFS (185.55s)
--- PASS: TestAccDataFactoryDatasetParquet_blobFSDynamicPath (273.66s)
PASS
ok      github.com/hashicorp/terraform-provider-azurerm/internal/services/datafactory   273.677s
```

### CODE REVIEW

File: internal/services/datafactory/data_factory_dataset_parquet_resource.go [hunk 1]
```diff
@@ -76,12 +76,54 @@ func resourceDataFactoryDatasetParquet() *pluginsdk.Resource {
 				},
 			},
 
+			// Parquet Specific Field, one option for 'location'
+			"azure_blob_fs_location": {
+				Type:          pluginsdk.TypeList,
+				MaxItems:      1,
+				Optional:      true,
+				ConflictsWith: []string{"azure_blob_storage_location", "http_server_location"},
```
Comment (stephybun) [inline internal/services/datafactory/data_factory_dataset_parquet_resource.go:84]: Can we change this to an `ExactlyOneOf`
```suggestion
				ExactlyOneOf: []string{"azure_blob_fs_location", "azure_blob_storage_location", "http_server_location"},
```

File: internal/services/datafactory/data_factory_dataset_parquet_resource.go [hunk 2]
```diff
@@ -76,12 +76,54 @@ func resourceDataFactoryDatasetParquet() *pluginsdk.Resource {
...
+				},
+			},
+
 			// Parquet Specific Field, one option for 'location'
 			"azure_blob_storage_location": {
 				Type:          pluginsdk.TypeList,
 				MaxItems:      1,
 				Optional:      true,
-				ConflictsWith: []string{"http_server_location"},
+				ConflictsWith: []string{"azure_blob_fs_location", "http_server_location"},
```
Comment (stephybun) [inline internal/services/datafactory/data_factory_dataset_parquet_resource.go:126]: Same here
```suggestion
				ExactlyOneOf: []string{"azure_blob_fs_location", "azure_blob_storage_location", "http_server_location"},
```

File: internal/services/datafactory/data_factory_dataset_parquet_resource.go [hunk 3]
```diff
@@ -159,7 +201,7 @@ func resourceDataFactoryDatasetParquet() *pluginsdk.Resource {
 				Type:          pluginsdk.TypeList,
 				MaxItems:      1,
 				Optional:      true,
-				ConflictsWith: []string{"azure_blob_storage_location"},
+				ConflictsWith: []string{"azure_blob_fs_location", "azure_blob_storage_location"},
```
Comment (stephybun) [inline internal/services/datafactory/data_factory_dataset_parquet_resource.go:204]: Same here
```suggestion
				ExactlyOneOf: []string{"azure_blob_fs_location", "azure_blob_storage_location", "http_server_location"},
```

**Review by stephybun — CHANGES_REQUESTED** [review]
Thanks for this PR @bianyifan. 

I left some minor comments in-line. We also need to update the docs for this resource, but once that's done we can take another look through.

Thanks!

**Review by stephybun — CHANGES_REQUESTED** [review]
There's a test failure for the delimited text data set which uses the same expand function

```
------- Stdout: -------
=== RUN   TestAccDataFactoryDatasetDelimitedText_blobFS
=== PAUSE TestAccDataFactoryDatasetDelimitedText_blobFS
=== CONT  TestAccDataFactoryDatasetDelimitedText_blobFS
------- Stderr: -------
panic: interface conversion: interface {} is nil, not bool
goroutine 681 [running]:
github.com/hashicorp/terraform-provider-azurerm/internal/services/datafactory.expandDataFactoryDatasetAzureBlobFSLocation(0xc00381c7e0?)
  /opt/teamcity-agent/work/3337027aeff310bf/internal/services/datafactory/data_factory.go:429 +0x585
github.com/hashicorp/terraform-provider-azurerm/internal/services/datafactory.expandDataFactoryDatasetLocation(0x7d26160?)
  /opt/teamcity-agent/work/3337027aeff310bf/internal/services/datafactory/data_factory.go:361 +0xb5
github.com/hashicorp/terraform-provider-azurerm/internal/services/datafactory.resourceDataFactoryDatasetDelimitedTextCreateUpdate(0x0?, {0x66c8bc0?, 0xc005ec8000?})
  /opt/teamcity-agent/work/3337027aeff310bf/internal/services/datafactory/data_factory_dataset_delimited_text_resource.go:345 +0x4a5
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0x7d46980?, {0x7d46980?, 0xc0046a67e0?}, 0xd?, {0x66c8bc0?, 0xc005ec8000?})
  /opt/teamcity-agent/work/3337027aeff310bf/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/resource.go:766 +0x178
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc0032ee700, {0x7d46980, 0xc0046a67e0}, 0xc003af56c0, 0xc0039a7500, {0x66c8bc0, 0xc005ec8000})
  /opt/teamcity-agent/work/3337027aeff310bf/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/resource.go:909 +0xa7e
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc0038eaf00, {0x7d46980?, 0xc0046a6720?}, 0xc003ae7310)
  /opt/teamcity-agent/work/3337027aeff310bf/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/grpc_provider.go:1060 +0xe8d
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc0050275e0, {0x7d46980?, 0xc004539ef0?}, 0xc0039fae70)
  /opt/teamcity-agent/work/3337027aeff310bf/vendor/github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server/server.go:859 +0x574
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x7116ce0?, 0xc0050275e0}, {0x7d46980, 0xc004539ef0}, 0xc0039fae00, 0x0)
  /opt/teamcity-agent/work/3337027aeff310bf/vendor/github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:467 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc001785860, {0x7d57a60, 0xc002c26ea0}, 0xc004557680, 0xc001f596b0, 0xc8ac138, 0x0)
  /opt/teamcity-agent/work/3337027aeff310bf/vendor/google.golang.org/grpc/server.go:1376 +0xdd2
google.golang.org/grpc.(*Server).handleStream(0xc001785860, {0x7d57a60, 0xc002c26ea0}, 0xc004557680, 0x0)
  /opt/teamcity-agent/work/3337027aeff310bf/vendor/google.golang.org/grpc/server.go:1753 +0xa36
google.golang.org/grpc.(*Server).serveStreams.func1.1()
  /opt/teamcity-agent/work/3337027aeff310bf/vendor/google.golang.org/grpc/server.go:998 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
  /opt/teamcity-agent/work/3337027aeff310bf/vendor/google.golang.org/grpc/server.go:996 +0x18c
```

**Comment by bianyifan** [comment]
Thanks for catching this! Just updated to make `data_factory_dataset_delimited_text` supports dynamic expressions in `azure_blob_fs_location` as well.

```
$ make acctests SERVICE='datafactory' TESTARGS='-run=TestAccDataFactoryDatasetDelimitedText_blobFS' TESTTIMEOUT=60m
==> Checking that code complies with gofmt requirements...
==> Checking that Custom Timeouts are used...
==> Checking that acceptance test packages are used...
TF_ACC=1 go test -v ./internal/services/datafactory -run=TestAccDataFactoryDatasetDelimitedText_blobFS -timeout 60m -ldflags="-X=github.com/hashicorp/terraform-provider-azurerm/version.ProviderVersion=acc"
=== RUN   TestAccDataFactoryDatasetDelimitedText_blobFS
=== PAUSE TestAccDataFactoryDatasetDelimitedText_blobFS
=== CONT  TestAccDataFactoryDatasetDelimitedText_blobFS
--- PASS: TestAccDataFactoryDatasetDelimitedText_blobFS (241.38s)
PASS
ok      github.com/hashicorp/terraform-provider-azurerm/internal/services/datafactory   241.397s
$ make acctests SERVICE='datafactory' TESTARGS='-run=TestAccDataFactoryDatasetParquet_blobFS' TESTTIMEOUT=60m
==> Checking that code complies with gofmt requirements...
==> Checking that Custom Timeouts are used...
==> Checking that acceptance test packages are used...
TF_ACC=1 go test -v ./internal/services/datafactory -run=TestAccDataFactoryDatasetParquet_blobFS -timeout 60m -ldflags="-X=github.com/hashicorp/terraform-provider-azurerm/version.ProviderVersion=acc"
=== RUN   TestAccDataFactoryDatasetParquet_blobFS
=== PAUSE TestAccDataFactoryDatasetParquet_blobFS
=== RUN   TestAccDataFactoryDatasetParquet_blobFSDynamicPath
=== PAUSE TestAccDataFactoryDatasetParquet_blobFSDynamicPath
=== CONT  TestAccDataFactoryDatasetParquet_blobFS
=== CONT  TestAccDataFactoryDatasetParquet_blobFSDynamicPath
--- PASS: TestAccDataFactoryDatasetParquet_blobFS (188.82s)
--- PASS: TestAccDataFactoryDatasetParquet_blobFSDynamicPath (261.98s)
PASS
ok      github.com/hashicorp/terraform-provider-azurerm/internal/services/datafactory   261.995s
```

**Review by stephybun — APPROVED** [review]
Thanks @bianyifan, tests are looking green. LGTM 👍

**Pull request merged**